packages/type-storage/src/fromMetadata.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright 2017-2018 @polkadot/storage authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import Metadata from '@polkadot/api-codec/Metadata';
+
+import { createFunction } from './utils/createFunction';
+import { ModuleStorage, Storage } from './types';
+
+/**
+ * Sets the 1st letter of a string to lowercase.
+ *
+ * @param s - The string to lower first letter.
+ */
+function lowerFirstLetter (s: string) {
+  return s.charAt(0).toLowerCase() + s.slice(1);
+}
+
+/**
+ * Extend a storage object with the storage modules & module functions present
+ * in the metadata.
+ *
+ * @param storage - A storage object to be extended.
+ * @param metadata - The metadata to extend the storage object against.
+ */
+export const fromMetadata = (storage: Storage, metadata: Metadata) => {
+  metadata.modules.forEach(moduleMetadata => {
```

Comment:
(nitpick) I know all the cool kids do it, but not a huge fan of the naked arguments. (So really have steered clear of them in all the polkadot-js repos)

Reply:
"naked argument" -> You mean adding explicit type here?